### PR TITLE
[MPDX-9391] - Adds status bar for spouse

### PR DESCRIPTION
**Pull request by wjames111**
## Description

When one spouse is over their cap but their spouse is under, we should show them their spouses ASR and how much room they have until their cap to show how they can submit it

This will likely show on the show "split funds with spouse" model

Jira ticket available [here](https://jira.cru.org/browse/MPDX-9391)

## Testing

- Impersonate Joel Hurdle
- Go to ASR page
- Add 10,000 to the first field
- Click submit
- Check the spouse bar shows

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels (_Add the label "Preview" to automatically create a preview environment_)
- [x] I have run the Claude Code `/pr-review` command locally and fixed any relevant suggestions
- [ ] I have requested a review from another person on the project
- [x] I have tested my changes in preview or in staging
- [x] I have cleaned up my commit history

### CODE REVIEW

**Review by wjames111 — COMMENTED** [review]
## Multi-Agent Code Review — PR #1716

**Verdict:** APPROVED WITH SUGGESTIONS  
**Risk:** LOW (3/10) · **Reviewer level:** ANY  
**Agents:** Architecture, Testing, Standards, UX, Financial Reporting (standard mode)

No blockers. 5 medium-priority suggestions and 3 low-priority nits — all posted as inline comments below.

### Top finding (4/5 agents agreed)
`t('{{totalFormatted}} / {{capFormatted}}', …)` is a pure-interpolation translation key — translators see `{{a}} / {{b}}` with no context and cannot re-order for RTL locales. Drop `t()` (it's numeric content, not prose) or use a semantic key like `t('{{total}} of {{cap}}', …)`.

### Cross-cutting observation
The new `spouseRemainingCap` clamps at 0 (`Math.max(0, cap − total)`), while the user-side equivalent `remainingInMaxAllowable` in `TotalSalaryRequested.tsx:40` is unclamped and can go negative. Either lift a shared `remainingCap` into `useSalaryCalculations` so both sides match, or document the asymmetry as intentional.

### Consistent with existing patterns (not flagged)
- Hardcoded `const currency = 'USD'` — matches 8 sibling files in the ASR feature.
- `as unknown as ReturnType<typeof useAdditionalSalaryRequest>` cast in tests — matches 11 sibling test files.

### How to respond
- **Fix** a finding → push a commit
- **Disagree** with any finding below severity 7 → reply `/dismiss: <reason>` on that comment
- **Re-review** after fixes → re-run `/quality:agent-review`

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx [hunk 1]
```diff
@@ -10,21 +16,98 @@ export const SplitCapSubContent: React.FC<SplitCapSubContentProps> = ({
...
+              justifyContent: 'space-between',
+            }}
+          >
+            <Typography variant="body1">
+              {t("{{spouseName}}'s Salary / Max Allowable Salary", {
+                spouseName,
+              })}
+            </Typography>
+            <Typography>
+              {t('{{totalFormatted}} / {{capFormatted}}', {
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx]: [Medium] **Pure-interpolation `t()` key.** The key `'{{totalFormatted}} / {{capFormatted}}'` has no translatable text — translators see only `{{a}} / {{b}}` with no context, and RTL/locale-specific reordering is impossible. Four of five review agents flagged this.

```suggestion
            <Typography>{`${totalFormatted} / ${capFormatted}`}</Typography>
```

Alternatively, use a semantic key: `t('{{total}} of {{cap}}', { total: totalFormatted, cap: capFormatted })`.

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx [hunk 2]
```diff
@@ -10,21 +16,98 @@ export const SplitCapSubContent: React.FC<SplitCapSubContentProps> = ({
...
+                totalFormatted,
+                capFormatted,
+              })}
+            </Typography>
+          </Box>
+          <LinearProgress
+            variant="determinate"
+            value={progressValue}
+            color="primary"
+          />
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx:95]: [Medium] **`LinearProgress` has no accessible name.** A screen reader announces "progressbar, 84" with no context — the preceding `Typography` is visually associated but not programmatically linked. Add `aria-labelledby` (pointing at an `id` on the header Typography) and `aria-valuetext` so the announcement includes the currency amounts:

```tsx
<Typography id="spouse-cap-label" variant="body1">
  {t("{{spouseName}}'s Salary / Max Allowable Salary", { spouseName })}
</Typography>
...
<LinearProgress
  variant="determinate"
  value={progressValue}
  color="primary"
  aria-labelledby="spouse-cap-label"
  aria-valuetext={`${totalFormatted} of ${capFormatted}`}
/>
```

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx [hunk 3]
```diff
@@ -10,21 +16,98 @@ export const SplitCapSubContent: React.FC<SplitCapSubContentProps> = ({
...
+        process.
+      </Trans>
+      {showBar && (
+        <Box sx={{ mt: 2 }}>
+          <Box
+            sx={{
+              mb: 1,
+              display: 'flex',
+              justifyContent: 'space-between',
+            }}
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx:77]: [Medium] **Flex row has no responsive wrap.** With a longer spouse name or localized label text (German, Spanish), the two `Typography` children will collide or clip on narrow viewports. Add `flexWrap: 'wrap'` and a small `gap`:

```suggestion
          <Box
            sx={{
              mb: 1,
              display: 'flex',
              justifyContent: 'space-between',
              flexWrap: 'wrap',
              gap: 1,
            }}
          >
```

Same applies to the footer row at line 99.

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx [hunk 4]
```diff
@@ -10,21 +16,98 @@ export const SplitCapSubContent: React.FC<SplitCapSubContentProps> = ({
...
+    spouseTotalAnnualSalary,
+    spouseIndividualCap,
+    spouseRemainingCap,
+  } = useSalaryCalculations({ values });
+
+  const showBar =
+    spouseIndividualCap !== null &&
+    spouseIndividualCap > 0 &&
+    spouseTotalAnnualSalary !== null &&
+    spouseRemainingCap !== null;
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.tsx]: [Suggestion] **`showBar` triple-null-check is redundant with the hook's contract.** `useSalaryCalculations` always returns all three spouse fields as a group (null-together or number-together). Simpler:

```suggestion
  const showBar =
    spouseIndividualCap !== null && spouseIndividualCap > 0;
```

Inside the `showBar` branch, TypeScript will need a local const for the other two values since the narrowing isn't transitive — acceptable given the hook contract.

File: src/components/Reports/AdditionalSalaryRequest/Shared/useSalaryCalculations.ts [hunk 5]
```diff
@@ -138,6 +144,11 @@ export const useSalaryCalculations = ({
       (exceedsCap && (!isMarried || spouseAtCap || spouseExceedsCap)) ||
       (userAtCap && isMarried && spouseExceedsCap);
 
+    const hasSpouseCap = isMarried && spouseIndividualCap !== null;
+    const spouseRemainingCap = hasSpouseCap
+      ? Math.max(0, spouseIndividualCap - spouseTotalAnnualSalary)
+      : null;
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/Shared/useSalaryCalculations.ts:160]: [Medium] **Duplicate remaining-cap logic with divergent semantics vs user-side.** This PR adds `spouseRemainingCap = Math.max(0, cap - total)` (clamped), but the user-side `remainingInMaxAllowable` in `src/components/Reports/AdditionalSalaryRequest/SubmitModalAccordions/TotalSalaryRequested/TotalSalaryRequested.tsx:40` is unclamped and can go negative. Per the Financial Reporting rules ("Goal math must match across UI layers"), unify them — either:

1. Add a symmetric `remainingCap: Math.max(0, individualCap - totalAnnualSalary)` to this hook and refactor `TotalSalaryRequested.tsx` to consume it, or
2. Document the intentional asymmetry (e.g., user-side uses signed remaining to drive a different display).

No live user-visible bug today — flagging to prevent drift.

File: src/components/Reports/AdditionalSalaryRequest/Shared/useSalaryCalculations.ts [hunk 6]
```diff
@@ -138,6 +144,11 @@ export const useSalaryCalculations = ({
       (exceedsCap && (!isMarried || spouseAtCap || spouseExceedsCap)) ||
       (userAtCap && isMarried && spouseExceedsCap);
 
+    const hasSpouseCap = isMarried && spouseIndividualCap !== null;
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/Shared/useSalaryCalculations.ts:150]: [Medium] **`spouseIndividualCap` null-vs-0 ambiguity.** The JSDoc on the new field says `null` when no spouse, but the pre-existing `?? 0` at line 76 also returns `0` when spouse exists without `spouseCalculations`. Then `hasSpouseCap = isMarried && spouseIndividualCap !== null` is `true` (0 !== null), so the hook returns `spouseIndividualCap: 0` — UI mitigates with the `> 0` guard, so no live bug, but the contract is fuzzy. Tighten with:

```ts
const spouseIndividualCap = spouse
  ? (requestData?.latestAdditionalSalaryRequest?.spouseCalculations
      ?.currentSalaryCap ?? null)
  : null;
```

Downstream `!== null` checks already cover this. Also verify `spouseTotalAnnualSalary` semantics under the new invariant (when spouse exists but calculations are missing, you likely want it to return `null` too, in parallel with cap).

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.test.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+    expect(
+      getByText(/Please make adjustments to your request/),
+    ).toBeInTheDocument();
+    expect(queryByRole('progressbar')).not.toBeInTheDocument();
+    expect(
+      queryByText(/Max Allowable Salary/),
+    ).not.toBeInTheDocument();
+  });
+});
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.test.tsx:172]: [Suggestion] **Missing edge case: `spouseIndividualCap === 0` should also hide the bar.** Currently only `spouseCap: null` is tested — but `showBar` also gates on `> 0`, and that branch is unexercised.

```tsx
it('does not render status bar when spouse cap is 0', () => {
  setupContext({ spouseCap: 0 });
  const { queryByRole } = renderSplitCapSubContent('Jane');
  expect(queryByRole('progressbar')).not.toBeInTheDocument();
});
```

File: src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.test.tsx [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    expect(
+      getByText(/Please make adjustments to your request/),
+    ).toBeInTheDocument();
+    expect(getAllByText(/Jane/).length).toBeGreaterThan(0);
+    expect(
+      getByRole('link', { name: 'Progressive Approvals' }),
+    ).toBeInTheDocument();
+  });
+
+  it('renders status bar header with spouse salary and cap amounts', () => {
```
Comment (wjames111) [inline src/components/Reports/AdditionalSalaryRequest/RequestPage/Helper/SplitCapSubContent.test.tsx:93]: [Suggestion] **`getAllByText(/Jane/).length > 0` is too loose** — it passes on any render containing "Jane" anywhere, including the new status bar (which also contains the spouse name). Assert a specific phrase from the Trans copy instead:

```suggestion
    expect(
      getByText(/Jane's maximum allowable salary/),
    ).toBeInTheDocument();
```
